components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/JDBCUserRecoveryDataStore.java

```diff
@@ -71,12 +59,15 @@ public void invalidate(UserRecoveryDataDO recoveryDataDO) throws IdentityExcepti
      */
     @Override
     public void invalidate(String userId, int tenant) throws IdentityException {
-        Connection connection = IdentityDatabaseUtil.getDBConnection();
+
         PreparedStatement prepStmt = null;
+        Connection connection = null;
         try {
+            connection = IdentityDatabaseUtil.getDBConnection();
             prepStmt = connection.prepareStatement(SQLQuery.INVALIDATE_METADATA);
-            prepStmt.setString(1, userId);
+            prepStmt.setString(1, userId.toLowerCase());
```

Comment:
Is it correct to do this always? What if the userid is case sensitive?

Reply:
This is the current behavior of the registry-based store. So we need to fix both in the future.

Reply:
Created https://github.com/wso2/carbon-identity-framework/issues/1830 to track this.